src/server/lib/http-request.js

```diff
@@ -0,0 +1,114 @@
+import originalFetch from "node-fetch";
+import AbortController from "node-abort-controller";
+import { log } from "../../lib";
+import { v4 as uuid } from "uuid";
+
+const requestWithRetry = async (
+  url,
+  { validateStatus, retries = retries || 0, timeout = 2000, ...props } = {}
+) => {
+  const requestId = uuid();
+
+  const retryDelay = () => {
+    const baseDelay = 50;
+    const randomDelay = Math.floor(Math.random() * (baseDelay / 2));
+    return baseDelay + randomDelay;
+  };
+
+  const statusValidator = status => {
```

Comment:
i know this is an upstream/different PR, but this looks like a lot of complexity for a single 'client' so far -- can you just use one?  If you really think all these validateStatus types are necessary, then document above the function signature so people know what they can do.
We don't have many functions where an argument could be multiple types (I know it's common enough in JS-land, in general) -- not sure we should start now.  The 'Spoke-y' code pattern for this kind of thing (I think this is also overkill) would be e.g. an `opts` argument that has possible `validStatuses`, `validationFunction`, `validStatus`

Reply:
@schuyler1d I'll narrow it down to 2 -- an array of result codes (for simple cases), and a function (for more complex cases).  It is already in an opts argument.

Reply:
And I'll change it to two names to avoid an overloaded parameter.

Reply:
b45dfd27f8f69ff139e1400762faa00fa0b822f1